libgiza/typed_dict.py

```diff
@@ -0,0 +1,95 @@
+"""
+Provides the basis for a more strictly typed dictionary that ensures all keys
+and values will be of a specific type, and will throw exceptions if keys,
+values, or pairs of keys and values are of the wrong type or fail validation.
+"""
+
+import abc
+
+
+class TypedDict(dict):
```

Comment:
This isn't Py3k-compatible. Unfortunately the syntax for metaclasses changed in an incompatible way, and the only solution is to use one of the shim libraries listed at http://python-future.org/compatible_idioms.html#metaclasses